### ISSUE TITLE
Unwanted keyboard focus on Modal body text

### ISSUE THREAD

**Issue by ashleylord8** [carbon-design-system/carbon#7453]
## Environment

> Operating system
Running Windows 10 VMware (on MacOS Catalina) &
Running directly on MacOS Catalina

> Browser
Firefox

> Assistive technology used to verify
JAWS

## Detailed description

> What version of the Carbon Design System are you using?
"carbon-components": "^10.20.0",
"carbon-components-react": "^7.20.0",

> What did you expect to happen?
No focus on `<Modal>` body text.

> What happened instead?
Within `<Modal>` component, text area receives unwanted focus. We try to remove keyboard focus on elements that don't require focus by adding `tabIndex="-1"` but no luck and guessing we shouldn't need to add it in the first place.

## Additional information

<img width="743" alt="Screen Shot 2020-12-14 at 10 07 24 AM" src="https://user-images.githubusercontent.com/20712108/102097711-4098c380-3df4-11eb-84d3-84980c98e155.png">

**Comment by emyarod** [comment]
I believe the modal body only receives focus when the `hasScrollingContent` prop is true. can you create a [reduced test case in Code Sandbox](https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox) for easier testing and debugging?
